src/tests/base/test_pricing_discount.py

```diff
@@ -1012,3 +1012,228 @@ def test_available_until(event, item):
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item.pk, None, Decimal('90.00'), False, False, Decimal('0.00')),
+        (item.pk, None, Decimal('90.00'), False, False, Decimal('0.00')),
+        (item.pk, None, Decimal('90.00'), False, False, Decimal('0.00')),
+    )
+    expected = (
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('100.00'),
+        Decimal('90.00'),
+        Decimal('0.00'),
+        Decimal('0.00'),
+    )
+
+    new_prices = [p for p, d in apply_discounts(event, 'web', positions)]
+    assert sorted(new_prices) == sorted(expected)
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_discount_subgroup_cheapest_n_min_count(event, item, item2):
+    # "For every 4 products, you get two free, but only of type item"
+    d1 = Discount(
+        event=event,
+        condition_min_count=4,
+        condition_all_products=False,
+        benefit_discount_matching_percent=100,
+        benefit_only_apply_to_cheapest_n_matches=2,
+        benefit_same_products=False,
+    )
+    d1.save()
+    d1.condition_limit_products.add(item)
+    d1.condition_limit_products.add(item2)
+    d1.benefit_limit_products.add(item)
+
+    positions = (
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item2.pk, None, Decimal('100.00'), False, False, Decimal('0.00')),
+        (item.pk, None, Decimal('90.00'), False, False, Decimal('0.00')),
+        (item.pk, None, Decimal('90.00'), False, False, Decimal('0.00')),
+        (item.pk, None, Decimal('90.00'), False, False, Decimal('0.00')),
```

Comment:
Also, I’d raise the price of `item` to something higher than 100, so the discount gets applied due to `benefit_limit_products` and not because of ` benefit_only_apply_to_cheapest_n_matches`.

```suggestion
        (item.pk, None, Decimal('110.00'), False, False, Decimal('0.00')),
        (item.pk, None, Decimal('110.00'), False, False, Decimal('0.00')),
        (item.pk, None, Decimal('110.00'), False, False, Decimal('0.00')),
```

Reply:
Ah sorry, the `benefit_limit_products` will be tested here properly as `benefit_only_apply_to_cheapest_n_matches` is 2, which would result in far more discounted items if it were not limited to item.

Reply:
I've made the test clearer with a n=1 and some comments